modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,237 @@
+
+.BackupLocation
+[options="header"]
+|===
+|Property|Type|Description
+
+|`velero`
+|*link:https://velero.io/docs/v1.9/api-types/backupstoragelocation/[velero.BackupStorageLocationSpec]
+|Location to store volume snapshots, as described in link:https://github.com/openshift/oadp-operator/blob/076f1495f6de65b0e96392cb6ef7cff66cf7defb/docs/config/bsl_and_vsl.md[Backup Storage Locations and Volume Snapshot Locations Customization].
+
+|`bucket`
+| [* link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#CloudStorageLocation[CloudStorageLocation] ]
+| [Technology Preview] Automates creation of a bucket at some cloud storage providers for use as a backup storage location.
+
+[IMPORTANT] +
+==== +
+The `bucket` parameter is a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them
+in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.
+
+For more information about the support scope of Red Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview].
+====
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#BackupLocation[Complete schema definitions for the type `BackupLocation`].
+
+.SnapshotLocation
+[options="header"]
+|===
+|Property|Type|Description
+
+|`velero`
+|*link:https://velero.io/docs/v1.9/api-types/volumesnapshotlocation/[velero.VolumeSnapshotLocationSpec]
+|Location to store volume snapshots, as described in link:https://github.com/openshift/oadp-operator/blob/master/docs/config/bsl_and_vsl.md[Volume Snapshot Location Specification].
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#SnapshotLocation[Complete schema definitions for the type `SnapshotLocation`].
+
+.ApplicationConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`velero`
+|*link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#VeleroConfig[VeleroConfig]
+|Defines the configuration for the Velero server.
+
+|`restic`
+|*link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#ResticConfig[ResticConfig]
+|Defines the configuration for the Restic server.
+|===
+
+link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#ApplicationConfig[Complete schema definitions for the type `ApplicationConfig`].
+
+.VeleroConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`featureFlags`
+|[] link:https://pkg.go.dev/builtin#string[string]
+|Defines the list of features to enable for the Velero instance.
+
+|`defaultPlugins`
+|[] link:https://pkg.go.dev/builtin#string[string]
+|Five types of default Velero plug-ins can be installed: `AWS`,`Azure`, `CSI`, `GCP`, and `OpenShift`.
```

Comment:
the plugin names should be lower cased, i.e, `openshift`, `aws`, `azure`, `csi`, `gcp`.
also missing additional default plugin option called `kubevirt`